processor/processor.go

```diff
@@ -284,7 +286,7 @@ func (proc *HandleT) Status() interface{} {
 func (proc *HandleT) Setup(
 	backendConfig backendconfig.BackendConfig, gatewayDB jobsdb.JobsDB, routerDB jobsdb.JobsDB,
 	batchRouterDB jobsdb.JobsDB, errorDB jobsdb.JobsDB, clearDB *bool, reporting types.ReportingI,
-	multiTenantStat multitenant.MultiTenantI,
+	multiTenantStat multitenant.MultiTenantI, transientSources transientsource.Service,
```

Comment:
Having to propagate transientSources inside proc, just to pass it to stash does not look very good. We can just inject it to errorDB, before we pass it here right?

If we do that, all the modifications in stash and proc won't be necessery.

Reply:
`proccessor` needs `transientSources` in order to perform sampleEvent filtering.
`stash` needs `transientSources` in order to perform event filtering as well.
`jobsdb` currently doesn't need  `transientSources`, it depends on `prebackup.Handlers`